pkg/build/strategies/dockerfile/dockerfile.go

```diff
@@ -128,6 +131,7 @@ func (builder *Dockerfile) CreateDockerfile(config *api.Config) error {
 	sourceDestDir := filepath.Join(getDestination(config), "src")
 	artifactsDestDir := filepath.Join(getDestination(config), "artifacts")
 	artifactsTar := sanitize(filepath.ToSlash(filepath.Join(defaultDestination, "artifacts.tar")))
+	imageScriptsDir := getImageScriptsDir(config)
```

Comment:
if "imageScriptsDir" came explicitly from the user (via ScriptsUrl with an image:// prefix), then we need to treat the scenario as "assembleProvided=true" (and run provided, save-artifacts provided, etc).  Basically the user has told us, with the highest level of precedence, "use the scripts from the image, i promise they are there".